publication/src/main/scala/hmda/publication/reports/disclosure/D51.scala

```diff
@@ -0,0 +1,155 @@
+package hmda.publication.reports.disclosure
+
+import java.util.Calendar
+
+import akka.NotUsed
+import akka.stream.scaladsl.Source
+import hmda.model.publication.reports.ApplicantIncomeEnum._
+import hmda.publication.reports._
+import hmda.model.publication.reports.ReportTypeEnum.Disclosure
+import hmda.model.publication.reports._
+import hmda.publication.reports.util.DateUtil._
+import hmda.publication.reports.util.DispositionType._
+import hmda.publication.reports.util.ReportUtil._
+import hmda.query.model.filing.LoanApplicationRegisterQuery
+
+import scala.concurrent.Future
+
+case class D51(
+  respondentId: String,
+  institutionName: String,
+  table: String,
+  reportType: ReportTypeEnum,
+  description: String,
+  year: Int,
+  reportDate: String,
+  msa: MSAReport,
+  applicantIncomes: List[ApplicantIncome],
+  total: List[Disposition]
```

Comment:
Looking at [the wiki page](https://github.com/cfpb/hmda-platform/wiki/Types-of-A&D-reports#disclosure), it looks like it would be safe to extract a couple of the fields that would be specific to disclosure reports: year, institution (name and respondent ID) and MSA.  I'm not sure about report date, but it seems like another common field.